### PR TITLE
fix: to avoid duplication of function variants in registry

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

**Review by srikrishnak — APPROVED** [review]

**Comment by scgkiran** [comment]
@EpsilonPrime @jacques-n 
Please review these changes

File: functions/dialect.go [hunk 1]
```diff
@@ -85,13 +83,14 @@ type withID interface {
 	ID() extensions.ID
 }
 
-func makeLocalFunctionVariantMap[T withID, V any](dialectFunctionInfos map[extensions.ID]*dialectFunctionInfo, getFunctionVariants func(string) []T, createLocalVariant func(T, *dialectFunctionInfo) *V) (map[FunctionName][]*V, error) {
+func makeLocalFunctionVariantMap[T withID, V any](dialectFunctionInfos map[extensions.ID]*dialectFunctionInfo, getFunctionVariants func(string) []T, createLocalVariant func(T, *dialectFunctionInfo) *V) (map[FunctionName][]*V, []*V, error) {
```
Comment (EpsilonPrime) [inline functions/dialect.go:86]: It'd be worth documenting this function's outputs.  It appears we are returning all local variants as a side effect of making a map of local function variants.  If they're the same data but represented two ways, we could use a new combined type (named a MapSlice or something local function specific) so we are only returning the data one way.

**Review by EpsilonPrime — APPROVED** [review]
Since these functions are internal it is okay to return two different things as part of your helper function if you'd like to revert back to that version.

File: functions/dialect_test.go [hunk 2]
```diff
@@ -1539,3 +1539,134 @@ scalar_functions:
...
+	require.NoError(t, c.Load(decimalUri, strings.NewReader(decimalYaml)))
+	funcRegistry := NewFunctionRegistry(&c)
+	localRegistry := getLocalFunctionRegistry(t, dialectYaml, funcRegistry)
+	allFunctions := funcRegistry.GetAllFunctions()
+
+	var fv []*LocalScalarFunctionVariant
+	fv = localRegistry.GetScalarFunctions(LocalFunctionName("sqrt"), 1)
+
+	expectedUris := []string{arithmeticUri, decimalUri}
+	expectedNames := []string{"sqrt:fp64", "sqrt:fp32", "sqrt:dec"}
```
[the remark this answers is not in the document]
Comment (scgkiran) [reply functions/dialect_test.go:1634]: as part of test, I didn't add that to the dialect in testcase.

**Comment by scgkiran** [comment]
> Since these functions are internal it is okay to return two different things as part of your helper function if you'd like to revert back to that version.

I am ok with either. With the current version the generic functions are more cleaner. So let's retain the current version.

File: functions/dialect_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply functions/dialect_test.go:1634]: I'm missing something.  There are four functions defined in the two files.  Their names are all distinct -- sqrt:i64, sqrt:fp64, sqrt:fp32, and sqrt:dec.  Shouldn't all of those names appear in this list?

[the remark this answers is not in the document]
Comment (scgkiran) [reply functions/dialect_test.go:1634]: Please see near line 1616 where test dialect is defined, I did **not** include `- i64` as one of the supported_kernel

[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply functions/dialect_test.go:1634]: Ah, got it.  Thanks.

**Review by EpsilonPrime — APPROVED** [review]

**Comment by scgkiran** [comment]
@EpsilonPrime can you please merge this PR? Thanks.

**Pull request merged**